### ISSUE TITLE
Rule request: no-implicit-null-return-types

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/tslint-microsoft-contrib#356]
*For teams with a large code base, turning on `--strictNullChecks` can take a while. Having rules that do part of the work would help incrementally enabling it.*

If a function may return null or undefined, its return type should reflect this.

Bad:

```typescript
class Person {
    private name;
    public setName(name: string) {
        this.name = name;
    }
    public getName(): string { // Might be undefined!
        return this.name;
    }
}
```

Better:

```typescript
class Person {
    private name;
    public setName(name: string) {
        this.name = name;
    }
    public getName(): string | undefined {
        return this.name;
    }
}
```

**Comment by JoshuaKGoldberg** [comment]
This is grander in scale than what a linter can do well. Addressing this would need access to the TypeScript language service.

I've started work on this in [TypeStat](https://github.com/JoshuaKGoldberg/TypeStat) instead.
